lib/coxir/model/entities/message/components.ex

```diff
@@ -0,0 +1,14 @@
+defmodule Coxir.Message.Components do
+  @moduledoc """
+  Work in progress.
+  """
+  use Coxir.Model, storable?: false
+  alias Coxir.Interaction.ComponentsData
+
+  @type t :: %Components{}
+
+  embedded_schema do
+    field(:type, :integer)
+    belongs_to(:components, ComponentsData)
```

Comment:
There is a compiler warning showing this is an invalid Ecto schema relation!

I presume we want this to be `embeds_many` instead?

Reply:
Also, why is there a separate `ComponentsData` module?

We could move the fields in `ComponentsData` to `Component` and then do:
```suggestion
    embeds_many(:components, Component)
```
And things will work as expected!